src/core/p5.Renderer2D.js

```diff
@@ -66,6 +67,9 @@ class Renderer2D extends Renderer {
     }
     this.scale(this._pixelDensity, this._pixelDensity);
 
+    if(!this.filterRenderer){
+      this.filterRenderer = new FilterRenderer2D(this);
```

Comment:
btw @davepagurek, my custom filterShader works for 2d mode. I need to add my filterRenderer here as well and in pixel.js as well

```js
      if (!this.filterRenderer) {
        this.filterRenderer = new FilterRenderer2D(this);
        this._renderer.filterRenderer = this.filterRenderer;
      }
```

any idea why this works only when I have initialized filterRenderer in both the places and I by removing any of them, it stops to work?

Reply:
I think `this` represents different things in both cases: in this file, it's a `p5.Renderer2D`, but in the pixels.js file, it's aninstance of `p5`. In that file, to access the one on the renderer, I think you need to check for `this._renderer.filterRenderer`.